clients/src/main/java/org/apache/kafka/common/Cluster.java

```diff
@@ -42,17 +43,32 @@
      * Create a new cluster with the given nodes and partitions
      * @param nodes The nodes in the cluster
      * @param partitions Information about a subset of the topic-partitions this cluster hosts
+     * @deprecated Use the Cluster constructor with 4 parameters
      */
+    @Deprecated
     public Cluster(Collection<Node> nodes,
                    Collection<PartitionInfo> partitions,
                    Set<String> unauthorizedTopics) {
-        this(false, nodes, partitions, unauthorizedTopics);
+        this(false, nodes, partitions, unauthorizedTopics, Collections.<String>emptySet());
```

Comment:
I wanted to avoid that because if we add an additional internal topic on the server, then we have a third possible behaviour: the deprecated constructor could be missing the new internal topic with an older client jar. It's not a strong reason, but it's typically easier to diagnose "all" or "nothing" cases.